### DISCUSSION TITLE
How to customise FILTER_DEFAULTS (e.g. for `FileField`)

### DISCUSSION THREAD

**Discussion by clement-escolano** [carltongibson/django-filter#1478] [Q&A] [answered]
When using django-filter with Django-Rest-Framework, I was confused because I could not use django-filter to filter on a `FileField` (I know the path of the file and want to get the item that has this file). When setting up a filter on a `FileField`, the following error occurs:

```
AssertionError: QuestionnaireFilter resolved field 'file' with 'exact' lookup to an unrecognized field type FileField. Try adding an override to 'Meta.filter_overrides'. See: https://django-filter.readthedocs.io/en/main/ref/filterset.html#customise-filter-generation-with-filter-overrides
```

<details>
  <summary>Code details</summary>
  
```python
class Questionnaire(models.Model):
    file = models.FileField(upload_to=questionnaire_path)
```
```python
class QuestionnaireFilter(FilterSet):
    class Meta:
        model = Questionnaire
        fields = ['file']
```
```python
class QuestionnaireViewset(viewsets.ModelViewSet):
    filterset_class = QuestionnaireFilter
```

</details>

I am confused because this seems a common use case and could not find any information about it except adding `filter_overrides` for every FilterSet I need.

<details>
<summary>Code details with filter_overrides</summary>

```python
class QuestionnaireFilter(FilterSet):
    class Meta:
        model = Questionnaire
        fields = ['file']
        filter_overrides = {
            models.FileField: {
                'filter_class': CharFilter,
                'extra': lambda f: {'lookup_expr': 'exact'},
            },
        }
```
</details>
Is this on purpose to not support the `FileField`?
Is there some way to make this filter_overrides globally to avoid adding it in a lot of filter sets?

Thank you and have a good day

**Comment by carltongibson** [comment, marked as answer]
Hi @clement-escolano. 

There's no `FileFilter`. 

> Is there some way to make this filter_overrides globally to avoid adding it in a lot of filter sets?

You can declare a base `FilterSet` subclass for your project. Inheritance of internal `Meta` classes gets sticky, so adjusting `FILTER_DEFAULTS` is likely the way to go. See how the `rest_framework.FilterSet` class does it: 

https://github.com/carltongibson/django-filter/blob/c906cf31498cc2d8ee915c7eae700d2b7f37c132/django_filters/rest_framework/filterset.py#L17-L25

I hope that helps.

**Reply by clement-escolano** [reply]
Alright thank you! I will follow the same pattern

**Reply by clement-escolano** [reply]
Out of curiosity, why there is no default filter for `FileField`?

**Reply by Real-Gecko** [reply]
Just saved my day :D
